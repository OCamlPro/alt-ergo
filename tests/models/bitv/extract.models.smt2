(set-logic BV)
(set-option :produce-models true)

(declare-const a (_ BitVec 2))
(declare-const b (_ BitVec 3))
(declare-const c (_ BitVec 6))
(declare-const z (_ BitVec 1))

(assert (= ((_ extract 0 0) c) #b1))
(assert (= ((_ extract 5 5) c) #b1))
(assert (= (concat (concat b z) a) c))

(assert (= ((_ extract 1 1) a) #b1))
(assert (= ((_ extract 1 0) b) #b11))
(assert (= z #b0))

(check-sat)
(get-model)
