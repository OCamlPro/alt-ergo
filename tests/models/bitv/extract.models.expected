
unknown
(
  (define-fun z () (_ BitVec 1) #b0)
  (define-fun a () (_ BitVec 2) #b11)
  (define-fun b () (_ BitVec 3) #b111)
  (define-fun c () (_ BitVec 6) #b111011)
)
